operator-common/src/main/java/io/strimzi/operator/common/operator/resource/StatusUtils.java

```diff
@@ -136,4 +128,43 @@ public static Condition getPausedCondition() {
                 .withStatus("True")
                 .build();
     }
+
+    /**
+     * Validate the Custom Resource. This should log at the WARN level (rather than throwing) if the resource can safely
+     * be reconciled (e.g. it merely using deprecated API).
+     *
+     * @param <T>               Custom Resource type
+     * @param <P>               Custom Resource spec type
+     * @param <S>               Custom Resource status type
+     *
+     * @param reconciliation    The reconciliation
+     * @param resource          The custom resource
+     *
+     * @throws InvalidResourceException if the resource cannot be safely reconciled.
+     *
+     * @return set of conditions
+     */
+    public static <T extends CustomResource<P, S>, P extends Spec, S extends Status> Set<Condition> validate(Reconciliation reconciliation, T resource) {
+        if (resource != null) {
+            Set<Condition> warningConditions = new LinkedHashSet<>(0); // LinkedHashSet is used to maintain ordering
+
+            ResourceVisitor.visit(reconciliation, resource, new ValidationVisitor(resource, LOGGER, warningConditions));
+
+            return warningConditions;
+        }
+
+        return Collections.emptySet();
+    }
+
+    /**
+     * Adds additional conditions to te status (this expects)
+     *
+     * @param status                            The Status instance where additonal conditions should be added
+     * @param unknownAndDeprecatedConditions    The Set with the new conditions
+     */
+    public static void addConditionsToStatus(Status status, Set<Condition> unknownAndDeprecatedConditions)   {
```

Comment:
is the variable name as `unknownAndDeprecatedConditions` still valid? This method was for adding warning conditions before but now it was renamed as a more general to add conditisions, so maybe the param could just be `conditions`?

Reply:
Good catch. I renamed the method to give it more generic name but not the param. Fixed.